### PR TITLE
Metadata tree table

### LINKED ISSUE #3370: Error message unsightly
If a field is not filled in, an internal name is displayed in the error message. If there are several tabs, the problematic tab is not navigated to.

Screenshot

**Goal:** The error message should display the field name as defined in the ruleset. If the dialog shows a tab in which there were no errors, but only errors in another tab, it should switch to one of the tabs in which errors occurred.

### LINKED ISSUE #3450: Mandatory fields are not marked
The metadata entry does not show which of the fields are mandatory.

Screenshot 1

Only after clicking "Save" will they be shown in red.

Screenshot 2

For comparison, in version 2, these were marked with an *.

**Goal:** An asterisk (*) is displayed behind the mandatory metadata fields.

### PR DESCRIPTION
**Pull request by IkramMaalej**
Fixes #3450
Fixes #3370

### CODE REVIEW

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataTreeTable.xhtml [hunk 1]
```diff
@@ -44,15 +44,21 @@
...
-                <p:outputLabel value="#{item.label}:" rendered="#{not item.undefined}"/>
-                <p:outputLabel value="#{item.label}: ⚠️" style="background-color: gold;"
-                               title="#{msgs['dataEditor.undefinedKey']}" rendered="#{item.undefined}" />
+                <!-- field label
+                The field ID must match the field's input attribute "#{item.input}", so that the outputLabel can be mapped to the field with the "for" attribute -->
+                
+                <p:outputLabel for="#{item.input eq 'dataTable' ? '' : item.input}"
+                               value="#{item.label}: #{item.required ? ' *' :''}"
+                               rendered="#{not item.undefined}"/>
+                <p:outputLabel value="#{item.label}: ⚠️"
```
Comment (markusweigelt) [inline Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataTreeTable.xhtml:53]: What is the last character here?

**Review by markusweigelt — CHANGES_REQUESTED** [review]
Great work and code looks good for me!

Issue 5550 is not completely fixed

> If the dialog shows a tab in which there were no errors, but only errors in another tab, it should switch to one of the tabs in which errors occurred.

image

In my case i navigate to "Title record link" tab and press save. The errors occurs in tab "Process data" but tab did not switched.

Issue 3450 is not completely fixed cause asterisk is missing on process title label.

image

File: Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataTreeTable.xhtml (hunk 1 above)
Comment (IkramMaalej) [reply Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataTreeTable.xhtml:53]: ` <p:outputLabel value="#{item.label}: ⚠️"
                               style="background-color: gold;"
                               title="#{msgs['dataEditor.undefinedKey']}"
                               rendered="#{item.undefined}"/>`
   
  If an imported metadata is not defined in the ruleset, its label is marked with this character  ` ⚠️`

Comment (markusweigelt) [reply Kitodo/src/main/webapp/WEB-INF/templates/includes/metadataTreeTable.xhtml:53]: Thanks i thought it was an incorrect character.

**Review by markusweigelt — APPROVED** [review]

**Pull request merged**